apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -1,311 +1,479 @@
-            "apollo-opentelemetry",
-            Some(env!("CARGO_PKG_VERSION")),
-            None,
-        );
-        let _prev_global_provider = global::set_tracer_provider(provider);
-
-        Ok(tracer)
     }
 
-    /// Create a client to talk to our spaceport and return an exporter.
-    pub(crate) fn build_exporter(&self) -> Result<Exporter, ApolloError> {
-        let collector = match self.spaceport_config.clone() {
-            Some(cfg) => cfg.collector,
-            None => DEFAULT_SERVER_URL.to_string(),
+    fn extract_http_data(&self, span: &SpanData) -> Http {
+        let method = match span
+            .attributes
+            .get(&Key::new("method"))
+            .map(|data| data.as_str())
+            .unwrap_or_default()
+            .as_ref()
+        {
+            "OPTIONS" => 1,
+            "GET" => 2,
+            "HEAD" => 3,
+            "POST" => 4,
+            "PUT" => 5,
+            "DELETE" => 6,
+            "TRACE" => 7,
+            "CONNECT" => 8,
+            "PATCH" => 9,
+            _ => 0,
         };
-        let graph = self.graph_config.clone();
-
-        tracing::debug!("collector: {}", collector);
-        tracing::debug!("graph: {:?}", graph);
+        let version = span
+            .attributes
+            .get(&Key::new("version"))
+            .map(|data| data.as_str())
+            .unwrap_or_default();
+        let headers = span
+            .attributes
+            .get(&Key::new("headers"))
+            .map(|data| data.as_str())
+            .unwrap_or_default();
+        let uri = Uri::from_str(
+            &span
+                .attributes
+                .get(&Key::new("uri"))
+                .map(|data| data.as_str())
+                .unwrap_or_default(),
+        )
+        .unwrap_or_default();
+        let mut request_headers: HashMap<String, Values> =
+            if let ForwardValues::None = &self.send_headers {
+                HashMap::new()
+            } else {
+                serde_json::from_str::<HashMap<String, Vec<String>>>(&headers)
+                    .unwrap_or_default()
+                    .into_iter()
+                    .map(|(header_name, value)| (header_name.to_lowercase(), Values { value }))
+                    .collect()
+            };
+
+        match &self.send_headers {
+            ForwardValues::Only(headers_to_keep) => {
+                request_headers.retain(|header_name, _v| headers_to_keep.contains(header_name));
+            }
+            ForwardValues::Except(skip_headers) => {
+                request_headers.retain(|header_name, _v| !skip_headers.contains(header_name));
+            }
+            ForwardValues::None | ForwardValues::All => {}
+        }
 
-        Ok(Exporter::new(collector, graph))
+        Http {
+            method,
+            host: uri.host().map(|h| h.to_string()).unwrap_or_default(),
+            path: uri.path().to_owned(),
+            request_headers,
+            response_headers: HashMap::new(),
```

Comment:
fwiw it doesn't look like AS sends response_headers either (I didn't realize that but makes sense I guess)... but you don't actually put anything in there either, so that's ok.

Reply:
fwiw i wish we sent response `content-length` somewhere for statistical analysis reasons!

Reply:
(we don't need to do that now)